### PR TITLE
Daniel - init code base

### PR DESCRIPTION
**Pull request by DanielRuizMobileRepository**
Daniel - init code base

### CODE REVIEW

File: dest_cocktail/src/components/HomePage/DrinkPreview.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  Image,
+  StyleSheet,
+  TouchableWithoutFeedback
+} from 'react-native'
+import {PropTypes} from 'prop-types'
+
+class DrinkPreview extends React.Component {
+
+  openDetailedDrink = () => {
+    const {idDrink} = this.props.data;
```
Comment (coreyakaplan) [inline dest_cocktail/src/components/HomePage/DrinkPreview.js]: Why not specify individual prop inputs to take and list them in proptypes?

File: dest_cocktail/src/components/HomePage/DrinkPreview.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  titleContainer: {
+    width: '50%'
+  },
+
+  title: {
+    fontSize: 24
+  },
+
+  thumbnail: {
+    width: 125,
```
Comment (coreyakaplan) [inline dest_cocktail/src/components/HomePage/DrinkPreview.js]: This is fine for a demo app, however in a production app that supports multiple phone sizes, hard coding sizes, margins, etc won't scale. Please provide an example of how you would write this to look equally proportioned/good on an iPhone 5, iPhone 10x, and iphone 7 Plus all with one code base.

File: dest_cocktail/src/components/HomePage/index.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  componentDidMount() {
+
+    getDrinksList()
+    .then(res => {
+      const {drinks} = res.data;
+
+      this.setState({drinks, filteredDrinks: drinks});
+    })
+    .catch(err => {
+      console.log(err);
```
Comment (coreyakaplan) [inline dest_cocktail/src/components/HomePage/index.js]: In a production application, this app would just appear to be frozen on error. Please provide an example of how you would provide feedback on an error to a user. Errors may include reachability for example.

File: dest_cocktail/src/components/HomePage/DrinkPreview.js (hunk 2 above)
Comment (DanielRuizMobileRepository) [reply dest_cocktail/src/components/HomePage/DrinkPreview.js]: In this case, I can create a scale function that will calculate a size of an element on every device individually, so all the elements will look alike.

Comment (coreyakaplan) [reply dest_cocktail/src/components/HomePage/DrinkPreview.js]: Great, please provide a code example of how you would do this.

File: dest_cocktail/src/components/HomePage/DrinkPreview.js (hunk 1 above)
Comment (DanielRuizMobileRepository) [reply dest_cocktail/src/components/HomePage/DrinkPreview.js]: Here’s a code example of simple scale function:

function scale(unit, size) {
    const dimensions = Dimensions.get(‘window’);


    const coefficient = size / dimensions[unit];
    const newSize = dimensions[unit] * coefficient;

    return newSize;
}

Also for that purpose there’s a lot of libraries that make your app responsible, and one of those libraries is ‘react-native-viewport-units’

File: dest_cocktail/src/components/HomePage/DrinkPreview.js (hunk 2 above)
Comment (DanielRuizMobileRepository) [reply dest_cocktail/src/components/HomePage/DrinkPreview.js]: Here’s a code example of simple scale function:

function scale(unit, size) {
const dimensions = Dimensions.get(‘window’);

const coefficient = size / dimensions[unit];
const newSize = dimensions[unit] * coefficient;

return newSize;
}

Also for that purpose there’s a lot of libraries that make your app responsible, and one of those libraries is ‘react-native-viewport-units’.

**Pull request closed without merging**